eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -393,4 +404,24 @@ class EclairImpl(appKit: Kit) extends Eclair {
     val sendPayment = SendPaymentRequest(amount, paymentHash, recipientNodeId, maxAttempts, externalId = externalId_opt, routeParams = Some(routeParams), userCustomTlvs = keySendTlvRecords)
     (appKit.paymentInitiator ? sendPayment).mapTo[UUID]
   }
+
+  override def signMessage(message: String): SignedMessage = {
+    val bytesToSign = SignedMessage.signedBytes(message)
+    val (signature, recoveryId) = appKit.nodeParams.keyManager.signDigest(bytesToSign)
+    SignedMessage(appKit.nodeParams.nodeId, message, (recoveryId + 31).toByte +: signature)
+  }
+
+  override def verifyMessage(message: String, recoverableSignature: ByteVector): VerifiedMessage = {
+    val signedBytes = SignedMessage.signedBytes(message)
+    val signature = ByteVector64(recoverableSignature.tail)
+    val recoveryId = recoverableSignature.head.toInt - 31
+    val pubKeyFromSignature = Crypto.recoverPublicKey(signature, signedBytes, recoveryId)
+    if (!pubKeyFromSignature.isValid)
+      VerifiedMessage(false, None)
+
+    if (Crypto.verifySignature(signedBytes, signature, pubKeyFromSignature))
+      VerifiedMessage(true, Some(pubKeyFromSignature))
+    else
+      VerifiedMessage(false, None)
```

Comment:
We try to always include the brackets, it doesn't use more lines and is less footgun-ish:

```suggestion
    if (Crypto.verifySignature(signedBytes, signature, pubKeyFromSignature)) {
      VerifiedMessage(true, Some(pubKeyFromSignature))
    } else {
      VerifiedMessage(false, None)
    }
```

Reply:
Done